File: Arcade/Intro/JavaScript/56_digitsProduct.js
function digitsProduct(product) {
    
    if (product === 0) {
        return 10;
    } else if (product === 1) {
        return 1;
    }
    
    var digits = [9,8,7,6,5,4,3,2];
    var min = '';
    
    for (var i = 0; i < digits.length; i++) {
        while (true) {
            if (product % digits[i] == 0) {
                product = product / digits[i];
                min = digits[i] + min;
            } else {
                break;
            }
        }
    }
    
    return product !== 1 ? -1 : Number.parseInt(min);
}

File: Arcade/Intro/C++/56_digitsProduct.cpp
int digitsProduct(int product) {
    for(int i=1;i<=99999;i++) {
        int p=1;
        string s = to_string(i);
        for(char c : s) p*=(c-'0');
        if(p==product) return atoi(s.c_str());
    }
    return -1;
}
